app/assets/stylesheets/views/_homepage.scss

```diff
@@ -216,108 +216,134 @@ body.homepage {
+          padding-right: 40px;
+          margin-right: -40px;
 
-        @include ie-lte(7){
-          padding-left: 0;
-        }
-        .js-enabled & {
-          width: 86%;
-          @include calc(width, "100% - 37px");
-        }
-      }
+          .search-input {
+            @include box-sizing(border-box);
+            @include appearance(none);
 
-      input.submit {
-        position: absolute;
-        z-index: 4;
-        right: 0;
-        top: 0;
-        width: 40px;
-        height: 40px;
+            position: relative;
+            height: 40px;
+            margin: 0;
 
-        border: 1px solid $black;
-        border-width: 0 0 0 1px;
-        @include ie-lte(7){
-          border-width: 1px;
+            border: 0;
+            outline-offset: 0;
+          }
         }
-        border-left-color: #222;
-        @include border-radius(0);
 
-        overflow: hidden;
-        text-indent: -5000px;
-        cursor: pointer;
+        .submit {
+          @include box-sizing(border-box);
+          float: left;
+          height: 40px;
+          width: 40px;
+        
+          border: none;
+          cursor: pointer;
 
-        color: $white;
-        background: $black image-url("homepage/search-button.png") no-repeat 2px 50%;
+          overflow: hidden;
+          text-indent: -5000px;
+          background: $black image-url("homepage/search-button.png") no-repeat 2px 50%;
+          
+          @include device-pixel-ratio() {
+            background-size: 52.5px 35px;
+            background-position: 115% 50%;
+          }
 
-        @include device-pixel-ratio() {
-          background-size: 52.5px 35px;
-          background-position: 115% 50%;
-        }
+          &:hover {
+            background-color: lighten($black, 5%);
+          }
+
+          &:focus {
+            z-index: 10;
+            position: relative;
+          }
 
-        @include ie-lte(7) {
-          background-image: none;
-          width: auto;
+          @include border-radius(0);
         }
 
-        &:hover {
-          background-color: lighten($black, 5%);
+        // If JS is enabled, and browser is not IE6 or IE7, position label
+        // as a placeholder
+        .js-enabled & {
+          position: relative;
+          background: $white;
+
+          // Absolutely position the label behind the input
+          label {
+            @include core-19($line-height: 40px, $line-height-640: 40px);
+            display: block;
+            position: absolute;
+            width: 100%;
+            left: 0;
+            top: 1px;
+            z-index: 5;
+
+            text-indent: 15px;
+
+            color: $secondary-text-colour;
+          }
+
+          // Input is transparent and sits above the label
+          .search-input {
+            position: relative;
+            z-index: 10;
+
+            // Internet Explorer does not play well with "empty" elements.
+            // url(#) somehow makes them work. ¯\_(ツ)_/¯
```

Comment:
It would be great to reference a URL explaining that further. I tried to google but Google only searches for "url", not "url(#)", although I put everything inside quotation marks.
I also don't understand the sentence. What are "empty" elements? And what exactly does the workaround fix?